### DISCUSSION TITLE
Is it possible to disable the rule NoUnusedFragments

### DISCUSSION THREAD

**Discussion by feedm3** [graphql-java/graphql-java#2595] [Q&A] [answered]
Hi there,

As the title already says, is it possible to disable the NoUnusedFragments rule?

Thanks in advance!

**Comment by jbellenger** [comment, marked as answer]
It is technically possible. You can do it by providing a PreparsedDocumentProvider to your GraphQL object. Your implementation of `PreparsedDocumentProvider.getDocument` can construct its own Validator object and completely skip invoking the provided `parseAndValidateFunction` callback.

This will basically void your warranty, and will use Internal graphql-java classes that are not part of the public api and might change between releases. Disabling this validation also violates rule [5.5.1.4: Fragments Must Be Used](https://spec.graphql.org/draft/#sec-Fragments-Must-Be-Used) of the graphql spec, which is a pretty bad idea and can break compatibility with other graphql ecosystem tools. I'd encourage you to reconsider going down this route.
